stan/math/prim/fun/trace_gen_inv_quad_form_ldlt.hpp

```diff
@@ -42,12 +42,11 @@ inline return_type_t<T1, T2, T3> trace_gen_inv_quad_form_ldlt(
     const Eigen::Matrix<T1, R1, C1> &D, const LDLT_factor<T2, R2, C2> &A,
     const Eigen::Matrix<T3, R3, C3> &B) {
   check_square("trace_gen_inv_quad_form_ldlt", "D", D);
-  if (D.size() == 0 && A.cols() == 0 && B.rows() == 0 && B.cols() == 0) {
-    return 0;
-  }
-
   check_multiplicable("trace_gen_inv_quad_form_ldlt", "A", A, "B", B);
   check_multiplicable("trace_gen_inv_quad_form_ldlt", "B", B, "D", D);
+  if (D.size() == 0 && A.cols() == 0) {
```

Comment:
This should be `(D.size() == 0 || A.size() == 0 || B.size() == 0)`.  If any of them are size zero, the result is going to either be empty or have all zero entries, so the trace is going to be 0.  There's another instance of this, too.

Reply:
Thanks for spotting this! I think it can be just `(D.size() == 0 || A.size() == 0)`, as either case implies `B.size() == 0`.